src/status_im/utils/datetime.cljs

```diff
@@ -97,6 +119,35 @@
           #(label :t/datetime-yesterday)
           #(label :t/datetime-today)))
 
+(defn timestamp->relative [ms]
+  (let [datetime       (from-long ms)
+        datetime-local (plus datetime time-zone-offset)
+        today          (minus (t/today-at-midnight) time-zone-offset)
+        yesterday      (minus today (days 1))
+        six-days-ago   (minus today (days 6))]
+    (cond
+      ;; Previous years.
+      (< (t/year datetime) (t/year today))
```

Comment:
consider putting these into helper functions and remove comments.
```
(defn previous-years? [datetime today datetimelocal]
(< (t/year datetime) (t/year today))
      (.format ^js (date-fmt) datetime-local))
```
same for below.

Reply:
Good suggestion, I'll do my best to see if this can be refactored. Just so you understand, the reason why I chose to keep things like this is because every subsequent check in the `cond` macro only works because of the previous checks.

For example, the correct way to check if a datetime is yesterday is not just `(before? datetime today)`. I hope you get it. I'll see what I can come up with, thanks.

Reply:
yeah I get that, I mean just use named functions for the checks rather than comments. You can still keep the function `defn timestamp->relative [ms]` as is

Reply:
It was quite more involved than a simple extraction, especially because I had to make sure the new implementation is correct. The new extracted functions are now useful on their own, and are correct, independent of the context they're being used, which was not the case of the original code. I also removed other superfluous timezone-related code that wasn't really necessary. I'm marking this thread as resolved :)